Cython/Build/Tests/TestRecythonize.py

```diff
@@ -1,212 +1,221 @@
-import shutil
 import os
+import shutil
 import tempfile
 import time
 
 import Cython.Build.Dependencies
 import Cython.Utils
 from Cython.TestUtils import CythonTest
 
+SAME = "The result of cytonization is the same"
+INCORRECT = "Incorrect cythonization"
+LINE_BEFORE_IMPLEMENTATION = '  /* "{filename}":{at_line}\n'
+VARS_LINE = '  /*--- Wrapped vars code ---*/\n'
 
-def fresh_cythonize(*args, **kwargs):
-    Cython.Utils.clear_function_caches()
-    Cython.Build.Dependencies._dep_tree = None  # discard method caches
-    Cython.Build.Dependencies.cythonize(*args, **kwargs)
 
 class TestRecythonize(CythonTest):
+    language_level = 3
+    dep_tree = Cython.Build.Dependencies.create_dependency_tree()
+
+    def clear_function_and_Dependencies_caches(self):
+        Cython.Utils.clear_function_caches()
+        Cython.Build.Dependencies._dep_tree = None  # discard method caches
 
     def setUp(self):
         CythonTest.setUp(self)
+        self.clear_function_and_Dependencies_caches()
         self.temp_dir = (
             tempfile.mkdtemp(
                 prefix='recythonize-test',
                 dir='TEST_TMP' if os.path.isdir('TEST_TMP') else None
             )
         )
+        self.src_dir = os.path.join(self.temp_dir, 'src')
+        os.mkdir(self.src_dir)
 
     def tearDown(self):
         CythonTest.tearDown(self)
+        self.clear_function_and_Dependencies_caches()
         shutil.rmtree(self.temp_dir)
 
-    def test_recythonize_pyx_on_pxd_change(self):
-
-        src_dir = tempfile.mkdtemp(prefix='src', dir=self.temp_dir)
-
-        a_pxd = os.path.join(src_dir, 'a.pxd')
-        a_pyx = os.path.join(src_dir, 'a.pyx')
-        a_c = os.path.join(src_dir, 'a.c')
-        dep_tree = Cython.Build.Dependencies.create_dependency_tree()
-
-        with open(a_pxd, 'w') as f:
-            f.write('cdef int value\n')
+    def write_to_file(self, path, text):
+        with open(path, "w") as f:
+            f.write(text)
+
+    def write_to_file_with_timestamp_update(self, path, text):
+        try:
+            timestamp_before_change = os.path.getmtime(path)
+        except OSError:  # not FileNotFoundError for compatibility
+            timestamp_before_change = .0
+
+        self.write_to_file(path, text)
+
+        # Make sure the file has a newer timestamp,
+        # otherwise cythonize does not work as expected
+        # on Linux-like systems #4245
```

Comment:
Since future readers of this code are very likely outside of the scope of github, any reference to a ticket should use the complete URL to look it up directly.
```suggestion
        # on Linux-like systems.
        # See https://github.com/cython/cython/issues/4245
```

Reply:
resolved [here](https://github.com/cython/cython/pull/4241/commits/a6d3d0a84a47d7f84fa3947dcc8a56d9e79dd0dc)

Reply:
stop ..  was it intended to remove the "Make sure the file has a newer timestamp,"?

Reply:
fully resolved [here](https://github.com/cython/cython/pull/4241/commits/99bb30a604daf58f9a2e2a32110bcb9d16f7b58c)